### DISCUSSION TITLE
Adding minifying images (and videos?) to upload.

### DISCUSSION THREAD

**Discussion by annardini** [payloadcms/payload#3927] [Feature Requests & Ideas]
I have been building a system using Payload and Next JS for a construction company. They provided some very large (but good quality) photos. I already have AWS Serverless Image Handler to scale photos but to reduce the costs on AWS and speed up loading times I have to minify the images using tinypng. I was thinking would it be a good idea to add the minifying facility to the upload functionality.

I think Sharp may have a feature built-in that can compress images but I am unsure if the quality downgrades or not? If anyone has an idea to create this functionality as a hook and then maybe a plugin that could also be an idea.

**Comment by MarkAtOmniux** [comment]
TinyPNG is just a lossy image compressor that removes certain colours from the image. You can achieve the same using sharp, webp, and the `lossy` or `near-lossless` options. You must include a value for `quality` if you want to enable these options.

https://sharp.pixelplumbing.com/api-output#webp
